compiler/damlc/tests/daml-test-files/SequentialLetSig.daml

```diff
@@ -0,0 +1,23 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- | Check that damlc preserves local type signatures
+-- in sequential let statements.
+module SequentialLetSig where
+
+preservesLocalTypeSigs : ()
+preservesLocalTypeSigs =
+    let f : Int -> Int -> Int
```

Comment:
I guess this fails due to ambiguity without the signature? Another option might be to add wrong signatures and make sure that it fails with a mismatch.

Reply:
Oooh, that's a good idea!

Reply:
Added a test!